### PR TITLE
Remove the guardian email constraint for junior division

### PR DESCRIPTION
**Pull request by devkonrad**
The changes inj this  PR allows a Junior Student to be a Guardian of a Beginner Student.

The details are in the related issue: #3165

### CODE REVIEW

**Comment by shaunxp20** [comment]
> allows a Junior Student to be a Guardian of a Beginner Student.

Hmm, I'm not sure this is what we want based on the title of #3165 - "Allow an email used to create a beginner account to be used as a parent email for another student". Maybe @dboyer can provide some insights here? 🤔 😄

**Comment by dboyer** [comment]
Thanks for checking on this.

When a user signs up for a junior or senior division account, the user (age 13-18) enters their own email address as the main contact for the account. However, they also have to provide the email address for a parent or guardian so that parent or guardian can sign the consent form for the user to participate.

Currently, the email for the parent or guardian cannot be used as the email for creating a beginner account. Beginner division accounts must be created by a parent since the girls are under age 13. The email address used to create a beginner division account should also be allowed to be entered as the parent/guardian email for a junior or senior division account.

The account used to create a junior or senior division account must be unique. It cannot be reused to create a beginner division account.

**Comment by devkonrad** [comment]
> Thanks for checking on this.
> 
> When a user signs up for a junior or senior division account, the user (age 13-18) enters their own email address as the main contact for the account. However, they also have to provide the email address for a parent or guardian so that parent or guardian can sign the consent form for the user to participate.
> 
> Currently, the email for the parent or guardian cannot be used as the email for creating a beginner account. Beginner division accounts must be created by a parent since the girls are under age 13. The email address used to create a beginner division account should also be allowed to be entered as the parent/guardian email for a junior or senior division account.
> 
> The account used to create a junior or senior division account must be unique. It cannot be reused to create a beginner division account.

@shaunxp20, hi! The error that @dboyer mentioned in the comment on #3165 was caused because the validation in "app/models/student_profile.rb" (modified in this PR) was applied to senior and junior accounts (no just to Senior Account). So i removed the validations for junior and left that one for seniors. Maybe there is something that i forgot. Any other idea? Thanks! :)

**Comment by shaunxp20** [comment]
I tested this one locally and noticed that:
- for a senior account, I cannot use my account email address for the parental consent email address
  - the form is re-rendered and an error message is displayed
- for a junior account, I cannot use my account email address for the parental consent email address
   - the form is re-rendered and **no** error message is displayed - I checked this behavior on QA, and it returns an error message, it seems something in this branch is changing this behavior, maybe someone can double-check me on this?

File: spec/features/allow_beginner_email_to_be_parent.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      expect(page).to have_content("Set your email and password")
+      
+      # Fill email and password
+      fill_in("email", with: "junior-student@email.com")
+      fill_in("password", with: "@student1234")
+  
+      # Submit the form
+      find(:css, '#formulate-global-13').click()
+  
+      # Testing for error message
```
Comment (shaunxp20) [inline spec/features/allow_beginner_email_to_be_parent.rb:60]: Thanks for writing a test for this! 👍 The comments aren't necessary though, the test steps are pretty self-explanatory as is. 😄

**Comment by devkonrad** [comment]
> I tested this one locally and noticed that:
> 
> * for a senior account, I cannot use my account email address for the parental consent email address
>   
>   * the form is re-rendered and an error message is displayed
> * for a junior account, I cannot use my account email address for the parental consent email address
>   
>   * the form is re-rendered and **no** error message is displayed - I checked this behavior on QA, and it returns an error message, it seems something in this branch is changing this behavior, maybe someone can double-check me on this?

Thanks for the catch @shaunxp20! I will take a deep look on this one. :)

**Comment by shaunxp20** [comment]
Closing this one since it was addressed in #3636.

**Pull request closed without merging**